x-pack/metricbeat/module/panw/system/power.go

```diff
@@ -0,0 +1,73 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package system
+
+import (
+	"encoding/xml"
+	"fmt"
+	"time"
+
+	"github.com/elastic/beats/v7/metricbeat/mb"
+	"github.com/elastic/beats/v7/x-pack/metricbeat/module/panw"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+)
+
+const powerQuery = "<show><system><environmentals><power></power></environmentals></system></show>"
+
+// getPowerEvents retrieves power-related events from a PAN-OS device.
+func getPowerEvents(m *MetricSet) ([]mb.Event, error) {
+
+	var response PowerResponse
+
+	output, err := m.client.Op(powerQuery, panw.Vsys, nil, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to execute operation: %w", err)
+	}
+
+	err = xml.Unmarshal(output, &response)
```

Comment:
Suggestion:

```
if len(output) == 0 {
    return nil, fmt.Errorf("empty response from power query")
}
```

Reply:
I don't think len(output) == 0 is possible given a nil err. The pango code will return any and all data from the response, so even if there are no results to unmarshal, then output will not be empty.